### PR TITLE
DIDDocument overview

### PR DESCRIPTION
**Pull request by coveloper**
This is just a basic overview.  It needs work still.  

It might make sense to combine the `DID` and `DIDDocument` pages?

### CODE REVIEW

File: book/section3/dids-and-diddocuments.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+A **DID** (*Decentralized Identifier*) is the canonical representation of a **DIDDocument**, a portable, compact hash, which can be passed around easily or stored to a database or blockchain. (ex. *did:prism:identifier123:hashOfDIDDocument123*)  A **DID** can be *resolved*, revealing the full, parsable JSON encoded **DIDDocument**.
+
+The spec for a `did:prism` DIDDocument can be found [here](https://github.com/input-output-hk/prism-did-method-spec/blob/main/w3c-spec/PRISM-method.md#did-documents).
+
+An Example **DID**:
+`
+did:prism:4a5b5cf0a513e83b598bbea25cd6196746747f065246f1d3743344b4b81b5a74:Cr4BCrsBElsKBmF1dGgwMRJRCglzZWNwMjU2azESBHRlc3QaOmtleTE6Ly8wMjM5MmYxNjc4NmNlNmQ0NzJlOGViNzA4ZWRjMmE3OTFmZGMxNzNkNjVkNTBhODNhMTk3N2I5ZmIwMmU0MjQSWwoGYXV0aDAyElEKCXNlY3AyNTZrMRIEdGVzdBo6a2V5MjovLzAyMzkyZjE2Nzg2Y2U2ZDQ3MmU4ZWI3MDhlZGMyYTc5MWZkYzE3M2Q2NWQ1MGE4M2ExOTc3YjlmYjAyZTQyNA
+`
```
Comment (robertocarvajal) [inline book/section3/dids-and-diddocuments.md:15]: Missing new line in here!

File: book/section3/dids-and-diddocuments.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    - `type`: The type of [Verification Method](https://www.w3.org/TR/did-core/#dfn-verification-method), ie [`Ed25519VerificationKey2020`](https://www.w3.org/community/reports/credentials/CG-FINAL-di-eddsa-2020-20220724/#ed25519verificationkey2020) or [JsonWebKey2020](https://www.w3.org/community/reports/credentials/CG-FINAL-lds-jws2020-20220721/#json-web-key-2020)
+- **Authentication Methods**:
+    -   `authentication`, `assertionMethod`, `keyAgreement`: Arrays of locations in the **Subject DID**, referenced in a **DID** + anchor format (`did:prism:1234#authentication0)
+    - *Author's note - Specify these in a more concrete way
+- **Services**: An array of `service` descriptions, including URLs that can be used to further interact with the **DID Subject**. These could be API endpoints, for messaging or file storage systems, but any remote service can be added to add value to the **DID**
+- **Controller**: An array of **DIDs** that are allowed to mutate the **DIDDocument**
+- **Proofs**: (*author's note: TBD) Contains cryptographic proofs that assert the validity and integrity of the **DID Document**, ensuring it has not been tampered with.
```
Comment (robertocarvajal) [inline book/section3/dids-and-diddocuments.md:65]: There is no proof section on example resolved DID

File: book/section3/dids-and-diddocuments.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    "id": "did:prism:123456789abcdefghi#messaging",
+    "type": "DIDCommMessaging",
+    "serviceEndpoint": "https://example.com/endpoint"
+}]
+```
+
+Let's look at the components of a DID Document:
+
+- **DID**: The **DID** of the **Subject** described by the **DIDDocument**
+- **Context**: Labled as `@context`, this is an array of context specs for the **DIDDocument**.  The first element is usually [https://www.w3.org/ns/did/v1](https://www.w3.org/ns/did/v1) but any other standards used in the DIDDocument should be listed here, for example URIs to definitions for [JSONWebSignature](https://w3id.org/security/suites/jws-2020/v1) or [DIDComm2 Messaging](https://didcomm.org/messaging/contexts/v2) protocol.
```
Comment (robertocarvajal) [inline book/section3/dids-and-diddocuments.md:52]: There is no context tag in the provided example resolved DID

**Comment by robertocarvajal** [comment]
Generally it's good for a draft but just noted some missing content on the example DID payload.. I would say it's fine for a draft because you added some author's notes in there that makes that very clear.

File: book/section3/dids-and-diddocuments.md (hunk 1 above)
Comment (coveloper) [reply book/section3/dids-and-diddocuments.md:15]: added, thanks!

File: book/section3/dids-and-diddocuments.md (hunk 3 above)
Comment (coveloper) [reply book/section3/dids-and-diddocuments.md:59]: Added, thanks!

**Comment by coveloper** [comment]
> Generally it's good for a draft but just noted some missing content on the example DID payload.. I would say it's fine for a draft because you added some author's notes in there that makes that very clear.

I addressed your feedback, but also revised the text to be more clear.

**Pull request merged**